tasks/image-segmentation/src/train.py

```diff
@@ -0,0 +1,147 @@
+        "description": taskRun.description,
+
+        "input_description": "RGB image",
+        "input_shape": model.input_shape,
+
+        "output_description": "Segmentation mask",
+        "output_shape": model.output_shape
+    })
+
+
+def train(taskRun: TaskRun) -> None:
+    if taskRun.parameters["epochs"] is None:
+        raise RuntimeError("The number of epochs for training the model is not defined")
+    if taskRun.parameters["validationSplit"] is None:
+        raise RuntimeError("The percentage of data from the dataset to be used for validation during model training is not defined")
+
+    taskRun.createMetrics([
+        Metric.create("loss", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]]),
+        Metric.create("accuracy", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]], [0, 1]),
+        Metric.create("val_loss", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]]),
+        Metric.create("val_accuracy", "epoch", MetricType.int, "value", MetricType.float, [0, taskRun.parameters["epochs"]], [0, 1])
+    ])
+
+    taskRun.updateStatus(TaskRunStatus.inProgress, "Downloading dataset")
+    taskRun.dataset.download()
+
+    excludedClasses: list[str] = taskRun.parameters["excludedClasses"]
+    logging.info(f">> [Image Segmentation] Excluding classes: {excludedClasses}")
+    taskRun.dataset.classes.exclude(excludedClasses)
+
+    count, dataset = loadDataset(taskRun.dataset, taskRun)
+    trainCount, trainBatches, testCount, testBatches = createBatches(
+        dataset,
+        count,
+        taskRun.parameters["validationSplit"],
+        taskRun.parameters["batchSize"],
+        taskRun.parameters["imageSize"]
+    )
+
+    # + 1 because we also have a background class
+    classCount = len(taskRun.dataset.classes) + 1
+    model = UNetModel(classCount, taskRun.parameters["imageSize"])
+
+    taskRun.updateStatus(TaskRunStatus.inProgress, "Training the model")
+
+    epochs: int = taskRun.parameters["epochs"]
+
+    try:
+        history: History = model.fit(
+            trainBatches,
+            epochs = epochs,
+            steps_per_epoch = math.ceil(trainCount / taskRun.parameters["batchSize"]),
+            validation_steps = math.ceil(testCount / taskRun.parameters["batchSize"]),
+            validation_data = testBatches,
+            callbacks = [DisplayCallback(epochs)],
+            verbose = 0,
+            use_multiprocessing = True
+        )
+    except tf.errors.ResourceExhaustedError:
+        raise MemoryError(">> [Image Segmentation] Ran out of memory. Potential solutions: reduce batch size and image size parameters; simplify model architecture")
+
+    detect.run(taskRun, model, taskRun.dataset)
+
+    coretexModel = Model.createModel(
+        taskRun.generateEntityName(),
+        taskRun.id,
+        history.history["val_accuracy"][-1]  # gets the validation accuracy after last epoch
```

Comment:
Can this accuracy be replaced with accuracy calculated by the running the validation on the model immediately after training?